File: src/github.com/cucumber/gherkin3/dialect.go
package gherkin3

type GherkinDialect struct {
	language string
	name     string
	native   string
	keywords map[string][]string
}

func (g *GherkinDialect) FeatureKeywords() []string {
	return g.keywords["feature"]
}

func (g *GherkinDialect) ScenarioKeywords() []string {
	return g.keywords["scenario"]
}

func (g *GherkinDialect) StepKeywords() []string {
	result := g.keywords["given"]
	result = append(result, g.keywords["when"]...)
	result = append(result, g.keywords["then"]...)
	result = append(result, g.keywords["and"]...)
	result = append(result, g.keywords["but"]...)
	return result
}

func (g *GherkinDialect) BackgroundKeywords() []string {
	return g.keywords["background"]
}

func (g *GherkinDialect) ScenarioOutlineKeywords() []string {
	return g.keywords["scenarioOutline"]
}

func (g *GherkinDialect) ExamplesKeywords() []string {
	return g.keywords["examples"]
}

func (g *GherkinDialect) Language() string {
	return g.language
}

func (g *GherkinDialect) Name() string {
	return g.name
}

func (g *GherkinDialect) Native() string {
	return g.native
}

type GherkinDialectProvider interface {
	GetDialect(language string) *GherkinDialect
}

type GherkinDialectMap map[string]*GherkinDialect

func (g GherkinDialectMap) GetDialect(language string) *GherkinDialect {
	return g[language]
}

File: python/gherkin3/dialect.py
import io
import os
import json


DIALECT_FILE_PATH = os.path.join(
    os.path.dirname(__file__),
    'gherkin-languages.json'
    )

with io.open(DIALECT_FILE_PATH, 'r') as file:
    DIALECTS = json.load(file, encoding='utf-8')


class Dialect(object):

    @classmethod
    def for_name(cls, name):
        return cls(DIALECTS[name]) if name in DIALECTS else None

    def __init__(self, spec):
        self.spec = spec

    def feature_keywords(self):
        return self.spec['feature']

    def scenario_keywords(self):
        return self.spec['scenario']

    def scenario_outline_keywords(self):
        return self.spec['scenarioOutline']

    def background_keywords(self):
        return self.spec['background']

    def examples_keywords(self):
        return self.spec['examples']

    def given_keywords(self):
        return self.spec['given']

    def when_keywords(self):
        return self.spec['when']

    def then_keywords(self):
        return self.spec['then']

    def and_keywords(self):
        return self.spec['and']

    def but_keywords(self):
        return self.spec['but']
